cpp/arcticdb/util/trace.cpp

```diff
@@ -6,21 +6,62 @@
  */
 
 #include <arcticdb/util/trace.hpp>
+#include <sstream>
 
 #ifndef  _WIN32
 #include <cxxabi.h>
 #endif
 
+#ifdef ARCTICDB_COUNT_ALLOCATIONS
+#include <cpptrace/cpptrace.hpp>
+#endif
+
 namespace arcticdb {
 
-    std::string get_type_name(const std::type_info& ti){
+std::string get_type_name(const std::type_info& ti) {
 #ifndef _WIN32
-        char* demangled = abi::__cxa_demangle(ti.name(), nullptr, nullptr, nullptr);
-        std::string ret = demangled;
-        free(demangled);
-        return ret;
+    char *demangled = abi::__cxa_demangle(ti.name(), nullptr, nullptr, nullptr);
+    std::string ret = demangled;
+    free(demangled);
+    return ret;
 #else
-        return ti.name();
+    return ti.name();
 #endif
+}
+
+#ifdef ARCTICDB_COUNT_ALLOCATIONS
+
+std::string get_trace() {
+    return cpptrace::generate_trace(5, 10).to_string();
+}
+
+std::string_view removePrefix(std::string_view input, std::string_view prefix) {
+    auto pos = input.rfind(prefix);
+    if (pos != std::string_view::npos) {
+        return {input.data() + prefix.size(), input.size() - prefix.size()};
     }
+    return input;
+}
+
+std::string unwind_stack(int) {
+    return get_trace();
 }
+
+std::string unwind_stack(int max_depth) {
+    void *buffer[max_depth];
+    int num_frames = backtrace(buffer, max_depth);
+    char **symbols = backtrace_symbols(buffer, num_frames);
+
+    thread_local std::ostringstream oss;
+    for (int i = 0; i < num_frames; ++i) {
+        auto filtered = removePrefix(symbols[i], "/opt/arcticdb/arcticdb_link/python/arcticdb_ext.cpython-38-x86_64-linux-gnu.so");
```

Comment:
Can you drop a more general prefix? This stripping will only work for certain dev setups